tools/cldr-code/src/main/java/org/unicode/cldr/test/DisplayAndInputProcessor.java

```diff
@@ -1289,6 +1297,12 @@ private String normalizeWhitespace(String path, String value) {
             value = PLACEHOLDER_SPACE_BEFORE.matcher(value).replaceAll("\u00A0{");
         }
 
+        // Finally, replace remaining space combinations with most restrictive type CLDR-17233
+        // If we have NNBSP U+202F in combination with other spaces, keep just it
+        value = NNBSP_AND_OTHER_SPACES.matcher(value).replaceAll("\u202F");
+        // Else if we have NBSP U+00A0 in combination with other spaces, keep just it
+        value = NBSP_AND_SPACES.matcher(value).replaceAll("\u00A0");
```

Comment:
Thanks

Reply:
@macchiati This also now handles thin space among other spaces. And I made the handling of other spaces more generic (using \h instead of an explicit list). Also added a unit test.